### PR TITLE
Deprecate OpenTelemetry SPI.

### PR DESCRIPTION
**Pull request by anuraaga**
We have autoconfigure module for SDK customization and would like to see patterns related to custom SDKs before SPIing to much initially.

Hopefully this helps with #2022

### CODE REVIEW

File: api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java [hunk 1]
```diff
@@ -33,6 +26,7 @@
  * @see TracerProvider
  * @see ContextPropagators
  */
+@SuppressWarnings("deprecation") // Remove after deleting OpenTelemetry SPI
```
Comment (jkwatson) [inline api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:29]: Wait up. We're deleting the global altogether? I think I might have missed a memo. I'm assuming this is a mistake?

**Review by jkwatson — CHANGES_REQUESTED** [review]
I don't think we are planning on deprecating the global...

**Comment by bogdandrutu** [comment]
@jkwatson I think you need to agree with @anuraaga on this, because I pointed the same in other PRs (or I got the same impression that this is the intent) and I tried to block it, but could not hold on this because everyone seem to want this.

**Comment by jkwatson** [comment]
> @jkwatson I think you need to agree with @anuraaga on this, because I pointed the same in other PRs (or I got the same impression that this is the intent) and I tried to block it, but could not hold on this because everyone seem to want this.

The intent is absolutely to remove the SPI from here, but not the global. At least, that's my understanding.

File: api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java (hunk 1 above)
Comment (anuraaga) [reply api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:29]: This is a suppresswarnings, no a Deprecated :P

Comment (jkwatson) [reply api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java:29]: oh. yes. :disappear:

**Review by jkwatson — APPROVED** [review]
Thanks!

**Comment by jkwatson** [comment]
> I don't think we are planning on deprecating the global...

And I'm right! And, totally misread an annotation.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**